### ISSUE TITLE
Possible miscompiling in cl-python

### ISSUE THREAD

**Issue by kpoeck** [clasp-developers/clasp#712]
load the following
```lisp
(defclass lexer-bug (closer-mop:standard-generic-function)
  ((string :initarg :string :accessor lexer-string))
  (:metaclass closer-mop:funcallable-standard-class))

(defmethod call-lexer (version (me lexer-bug) op)
  (list version me op))

(defun generate-function (string yacc-version)
  (let (*lex-state*)
    (declare (special *lex-state*))
    (let ((lexer (apply #'make-instance 'lexer-bug :string string nil)))
      (flet ((call-lexer-forwarder (&optional op)
               (let ((*lex-state* lexer))
                 (call-lexer yacc-version lexer op))))
        (closer-mop:set-funcallable-instance-function lexer #'call-lexer-forwarder)
        lexer))))
````
Execute:
```lisp
COMMON-LISP-USER> (funcall  (generate-function "1+2" :boo) 23)

Condition of type: SIMPLE-ERROR
No applicable method for CALL-LEXER with arguments of types 
 (INTEGER 1121809188 1121809188)
 (INTEGER 1121809188 1121809188)
 (INTEGER 23 23).

Available restarts:
(use :r1 to invoke restart 1)
`````
So something goes wrong in  (call-lexer yacc-version lexer op), e.g. yacc-version lexer  are not correctly put in the closure

**Comment by Bike** [comment]
Oh, whoops. This is probably #739, which I just filed for completely unrelated reasons. (Here you set the funcallable instance function with `call-lexer-forwarder`, which is a closure over `lexer` and `yacc-version`.)

Since my bug has a simpler example and causation, I'm going to close this as a duplicate even though you noticed it first.

**Comment by kpoeck** [comment]
Fixed with the fix for #739
